articles/Intro to C++ - Part 3/index.md

```diff
@@ -0,0 +1,141 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/functions-in-cpp/
+title: Functions in C++
+description: Functions in C++, what are they? How do they work? In this article we will be going over one of the most fundamental idea in programming.
+author: Parampreet Singh
+date: 2020-08-30T00:00:00-12:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/functions-in-cpp/hero.jpg
+    alt: functions Function C++ cpp 
+---
+Let us continue from where we left off in the previous article [*Introduction to C++*](/engineering-education/intro-to-c-part1/). We started by going over the importance of C++ as a programming language, and where it is being used today. Most importantly, we tried to go over the basic concepts of the C++ language that any developer would need to get started. If you haven't had a chance to read it already, make sure you do to get a better understanding as we move forward. 
+<!--more-->
+
+## Functions in C++
+
+### **What are Functions?**
+
+Functions have a name and take zero or more parameters. The parameters are how information gets sent to the function. The function only cares about the order of the parameters. It doesn't know or care about the names of the variables in your program. 
+
+The only way to send information to a function is through parameters. And the only way a function can send information back is by returning the value.
+
+**Why do we need function?**
+
+* To write a task once that you'll use a bunch of times.
+* To break the task of writing a big program into smaller, more easily handleable pieces.
+* To create a library of operations for use in multiple programs
+
+Let's write a function that returns the largest of the three parameters:
+
+```C
+int max_of_three (int one, int two, int three) {
+
+//If one is greater than or equal to two and three: return one
+if (one>=two && one>=three)
+   return one;
+
+//If two is greater than or equal to one and three: return two
+else if (two>=one && two>=three)
+  return two;
+
+//If none of the above statements is true, then three would be the highest: return three
+else
+  return three;
+}
+```
+The function max_of_three would take three parameters and return the largest of the three parameters.
```

Comment:
max_of_three would take three parameters  -> `max_of_three` takes three parameters

Reply:
and return the largest -> and returns the largest